src/main/java/jsr223/cpython/PythonScriptEngine.java

```diff
@@ -69,12 +67,29 @@ public PythonScriptEngine() {
 
     }
 
+    private static synchronized GatewayServer startGatewayServer(EntryPoint entryPoint) {
+        GatewayServer gatewayServer = new GatewayServer(entryPoint, 0);
+        gatewayServer.start();
+        return gatewayServer;
+    }
+
     @Override
     public Object eval(String script, ScriptContext context) throws ScriptException {
+
+        EntryPoint entryPoint = new EntryPoint();
+
+        //Create the EntryPoint and start the gateway server
+        GatewayServer gatewayServer = startGatewayServer(entryPoint);
```

Comment:
Why not directly creating an entrypoint in the startGatewayServer method?
So it will look like : 
GatewayServer gatewayServer = startGatewayServer();

Reply:
because we need to use the bindings of the entryPoint in eval.